src/main/java/org/zalando/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -135,6 +143,13 @@ public StreamingResponseBody streamEvents(
             SubscriptionStreamer streamer = null;
             final SubscriptionOutputImpl output = new SubscriptionOutputImpl(response, outputStream);
             try {
+                if  (subscriptionDbRepository.getSubscription(subscriptionId).getEventTypes().stream()
+                        .map(etName -> floodService.isConsumptionBlocked(etName)).findFirst().orElse(false)) {
+                    response.setStatus(HttpStatus.TOO_MANY_REQUESTS.value());
```

Comment:
>> `due to the last experiences on live as a fast solution`
>
> ...So maybe :shipit: but revisit/iterate ideas soon,

It's 14d since that comment. At this point this pr has gone beyond a quick patch and is something more fundamental. I think we should challenge the fail-closed policy of stopping access when the rate limiter fails. There are two scenarios I'm thinking about:

 - traffic level is fine. The rate limiter fails - system unavailable

- traffic level is not fine. The rate limiter fails - system unavailable

I'm worried that the intent of the limiter is to help maintain availability in the latter case of excess traffic, but has consequences for the former where the core system is working under acceptable traffic conditions.  A fail open policy with an alarm/escalation avoids the former scenario where the compensating mechanism takes out the system unnecessarily.